resources/lib/tubecast/youtube/app.py

```diff
@@ -430,30 +451,53 @@ def __monitor_volume(self):
 class YoutubeListener(threading.Thread):
 
     def __init__(self, app, ssdp=True):
-        threading.Thread.__init__(self)
-        self.app = app
+        super(YoutubeListener, self).__init__(name="YoutubeListener")
+        self.app = app  # type: YoutubeCastV1
         self.stop = False
         self.ssdp = ssdp
+        self.r = None  # type: Optional[requests.Response]
+
+    def __read_cmd_lines(self, url):  # type: (str) -> Iterator[str]
+        with self.app.session.get(url, stream=True) as self.r:
+            try:
+                for line in self.r.iter_lines():
+                    if self.stop:
+                        break
+
+                    yield line
+            except requests.exceptions.ChunkedEncodingError:
+                # raised when we forcefully close the socket.
+                # If we don't want to stop though, this should raise.
+                if not self.stop:
+                    raise
 
     def _listen(self):
         logger.debug("Listening to youtube remote events...")
         self.app.ofs += 1
-        bind_vals = self.app.bind_vals
+        # copy dict to avoid race condition with __post_bind which sets RID to 1337.
```

Comment:
Behaviour is clear (to ensure thread safety). Please remove the huge block of text if possible :)